hazelcast/src/main/java/com/hazelcast/util/ServiceLoader.java

```diff
@@ -400,6 +360,10 @@ private boolean advance() {
             return false;
         }
 
+        private boolean isDuplicate(Class<?> candidate) {
+            return !alreadyProvidedClasses.add(candidate);
```

Comment:
I was just thinking about this - this would eliminate classes that are equal. That includes the check if the class was loaded by the same classloader. So this check only eliminates same classes loaded by different classloaders? So for example, we had one classloader A loading the service definitions from one JAR and a different classloader B loading the service defintions from a different JAR. If both of them have the same parent and delegate to the parent, this check would eliminate duplicate service definitions from different JARs?